iotconnect-sdk/nrf-layer-lib/src/iotconnect_mqtt.c

```diff
@@ -325,11 +325,19 @@ static void mqtt_evt_handler(struct mqtt_client *const c,
             printk("MQTT PUBLISH result=%d len=%d\n", evt->result, p->message.payload.len);
             err = get_event_payload(c, p->message.payload.len);
             if (err >= 0) {
-
-                if (config->data_cb) {
-                    config->data_cb(payload_buf, p->message.payload.len, p->message.topic.topic.utf8);
+                if(! strncmp(p->message.topic.topic.utf8,"$iothub/twin/res/",17)){
+                    if (config->twin_cb) {
+                        config->twin_cb(payload_buf, p->message.payload.len, p->message.topic.topic.utf8);
+                    } else {
+                        printk("Error: no mqtt twin_cb configured\n");
+                    }
+                } else{
+                    if (config->data_cb) {
+                        config->data_cb(payload_buf, p->message.payload.len, p->message.topic.topic.utf8);
                 } else {
-                    printk("Error: no mqtt data_cb configured\n");
+                        printk("Error: no mqtt data_cb configured\n");
+                }
+
```

Comment:
Can this be properly indented to avoid confusion of the double braces back to back.

Reply:
https://github.com/avnet-iotconnect/iotc-nrf-sdk/commit/2e1de91e2c0518b1d6b3767504b08939b5b282e7 

updated in the above latest commit.